bleak/agent.py

```diff
@@ -0,0 +1,27 @@
+import abc
+
+from .backends.device import BLEDevice
+
+
+class BaseBleakAgentCallbacks(abc.ABC):
```

Comment:
This needs a doc comment that includes an overview.

Some important points:
* Method implementations that `await` need to consider that an `asyncio.CancelledError` can be raised on any coroutine that is called.
* Include `.. versionadded:: unreleased` (will be updated with actual version on release).